packages/context/src/inject.ts

```diff
@@ -201,9 +203,32 @@ export namespace Getter {
 }
 
 /**
- * The function injected by `@inject.setter(key)`.
+ * The function injected by `@inject.setter(bindingKey)`.
  */
-export type Setter<T> = (value: T) => void;
+export interface Setter<T> {
```

Comment:
What is the reasoning for this change? Do you need an interface so that you can inherit from it?

Also the API docs seems little weird to me - it says `the function` but describes an interface.

Reply:
I changed them to interfaces to define `Setter.applyTemplates()`. Now they can be reverted.